src/transformers/models/kosmos2/processing_kosmos2.py

```diff
@@ -145,10 +173,28 @@ def __call__(
         if images is None and text is None:
             raise ValueError("You have to specify either images or text.")
 
+        # Temporary fix for "paddding_side" in init_kwargs
+        _ = self.tokenizer.init_kwargs.pop("padding_side", None)
```

Comment:
It's related to `AutoTokenizer` mapping, @yonigozlan can say a bit more :)

Reply:
@zucchini-nlp From what I’ve seen, some tokenizers accept `padding_side` as a call argument, while others don’t. But when you save weights and configs using a tokenizer loaded with `AutoTokenizer` and then reload them later, all possible init kwargs (including `padding_side`) get added to the tokenizer’s `init_kwargs`, even if they weren’t explicitly specified in the first place. So when merging the `tokenizer.init_kwargs` with the `output_kwargs`, if the tokenizer doesn’t support `padding_side` in its call function, it will cause an error.

Hopefully, that makes sense - it’s still a bit unclear for me too, to be honest. :)

Reply:
Not sure I understand correctly. The basic TextKwargs have `padding-side` so seems like it should not be causing errors and should be assigning a kwarg to be used later, when calling the tokenizer. If users don't pass anything it will be the default kwarg from init time

Reply:
I guess the main problem is that `padding_side` is included in the basic `TextKwargs` when some tokenizer encode function don't accept it as an argument such as `batch_encode_plus` for `PretrainedTokenizerFast`:
https://github.com/huggingface/transformers/blob/d6751d91c8f58cdeb35af6adae182d7dc90aa883/src/transformers/tokenization_utils_fast.py#L489-L510
So maybe it shouldn't be in TextKwargs at all? Do we have an example of a tokenizer that needs to set padding_side at call time rather than at init time? What do you think @molbap ?

Reply:
That is correct, no `padding_side` is used at call time it seems - might have been an oversight on my end to include it in the first place. If that's the case we can check it and make sure it is indeed not used, and if that's the case removing it should be doable without breaking BC

Reply:
I didn't find any instances of `padding_side` being set at call time in Transformers, so I don't think removing it will break anything :) .
I used this regex to look in the library: `(?=.*\bpadding_side\b)(?=.*\bprocessor\b)\s*(.*\S.*)`
Which looks for lines where "padding_side" and "processor" are both used (and ignores leading whitespaces to avoid duplicate results)

Reply:
Yes, afaik padding side currently can be set only as `tokenizer.padding_side="left"`. Not sure if it will used any time in future as call time argument, so I am for removing it